src/runtime/vm/vm.cc

```diff
@@ -143,8 +143,15 @@ PackedFunc VirtualMachine::GetFunction(const std::string& name,
       } else {
         auto it = inputs_.find(func_name);
         ICHECK(it != inputs_.end()) << "Input has not been set for function " << func_name;
-        const std::vector<ObjectRef>& func_args = it->second;
-        *rv = Invoke(func, func_args);
+        const std::vector<ObjectRef>& input_args = it->second;
+        if (set_outputs_enabled_.count(func_name) && set_outputs_enabled_[func_name]) {
+          ICHECK(outputs_.count(func_name))
+              << "Outputs have not been set for function " << func_name;
+          *rv = Invoke(func, input_args, outputs_[func_name]);
+          set_outputs_enabled_[func_name] = false;
```

Comment:
It depends on scenario of using output tensors allocated outside. I see two options: 1. In-place scenario: memory for output tensors is allocated once. Each new inference (invoke) writes result in this memory. 2. We should insert new inputs/outputs for each new infer. Just now the second scenario is implemented and in that case `outputs_` can be cleared. But in the first scenario we should store `outputs_`. What do you think should I support the first scenario instead of second one or both ones or keep the current one?

Reply:
The register file holds a reference to the outputs too. So if we clear `outputs_`, they will be released only when the register file is reset.

Reply:
I have done it due to the second scenario is valid in this case but It should be noted that the first one is not. Each invoke the register file is refilled, we need to keep outputs_ if we want to save result in the same memory. Perhaps the API for the first scenario should be thought out more deeply and implemented separately. Just now the second scenario works only.